### PR TITLE
Replace FHIR Base classes with v4.0.1

### LINKED ISSUE #5737: Current FHIR R4 classes missing properties and methods
### Describe the bug
Current build was Generated on FHIR v4.0.0 schema.
Current released spec is FHIR v4.3.0 however v4.0.1 is the last version we can build until the builder project is updated to reflect newer schema changes.
For current and near term development we need min FHIR v4.0.1.
I've also noted there are several undefined namespaces and class paths in several places on existing code. Unclear how these would present but need to resolve them while I'm in this code.

Plan:
- Build library for FHIR v4.0.1 using the Oct 2019 schema.
- Remove FHIR directory tree.
- Replace with new v4.0.1 using the same OpenEMR\FHIR namespace
- Refactor all FHIR services and REST Controllers for class path changes.
- Optimize

@bradymiller Once I get the PR up maybe you can help me test. Mainly for PHP 8+ warnings which for some reason I miss many on my Windows PHP compile.

Because we've recently introduced PKCE @adunsulag will need to rerun all the inferno tests anyway.

Besides a change in project size (40mb vs 14mb v4.0.0) and new properties/getters, class paths have also changed with new backbone and data requirement element paths. the domain class was moved under FHIRResource path.

### PR DESCRIPTION
**Pull request by sjpadgett**
- remove existing FHIR/R4 class directory with v4.0.0 content
- add back new FHIR/R4 with v4.0.1 classes

Fixes #5737

#### Short description of what this resolves
Update FHIR base classes to v4.0.1 from v4.0.0
Refactor all appropriate services/controllers for new version

### CODE REVIEW

**Comment by sjpadgett** [comment]
@bradymiller If you get a chance will you look into tests. I believe there are some PHP 8+ errors getting in the way or I've missed some pathing's.
If not I'll pick back up tomorrow...

**Comment by bradymiller** [comment]
@sjpadgett , would focus on the test fails in the `Test / PHP 7.4 - Apache - MariaDB 10.6 (pull_request)` run to avoid any php8 warning distractions. Looks like mostly parameter class typing issues. Note the automated testing is not very comprehensive and guessing inferno testing will unmask more of these issues :)

**Comment by bradymiller** [comment]
Also, can skip psr12 styling check in the R4 classes by placing the exception here:
https://github.com/openemr/openemr/blob/master/ci/phpcs.xml#L32-L42

**Comment by sjpadgett** [comment]
This is becoming a considerable task. The last commit will show the type of issues we're up against from the FhirPractitionerRestControllerTest.php. We'll need to go through all our controllers and services. I took care of validation but the FHIR parses in each controller will need to be addressed.

I'll leave this open for anyone wanting to tackle some of it but I need to move on and finish my questionnaire project where I'll bring in the v4.0.1 library separate. Then perhaps we can move existing over bit by bit.
This is very concerning because if you think this is bad, wait till R5.

File: src/Services/FHIR/FhirPatientService.php [hunk 1]
```diff
@@ -352,7 +354,7 @@ private function parseOpenEMRRaceRecord(FHIRPatient $patientResource, $race)
 
         $textExtension = new FHIRExtension();
         $textExtension->setUrl("text");
-        $textExtension->setValueString(new FHIRString($ombCategoryCoding->getDisplay()));
+        $textExtension->setValueString(new FHIRString($ombCategoryCoding->getDisplay()->getValue()->getValue()));
```
Comment (sjpadgett) [inline src/Services/FHIR/FhirPatientService.php:357]: Need to add some more helpers to Questionnaire Traits to validate and get value properties too make life easier...

**Comment by sjpadgett** [comment]
I've abandoned this for now. I reworked my questionnaire project to use current R4 leaving me a little disappointed I couldn't move us past the v4.0.0 version.

Though this wouldn't be too hard to finish in terms of understanding the differences between versions, I think if effort is going to go into changing versions then we should go straight to R5. This will mean @adunsulag and I will need to help Carbone update his project and figure out how we will handle supporting two version for our APIs. e.g. routing based on called FHIR version. yuck!

**Pull request closed without merging**